tgui/packages/tgui/interfaces/Radio/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { useBackend } from "../../backend";
+import {
+  AnimatedNumber,
+  Box,
+  Button,
+  Knob,
+  LabeledList,
+  Section,
+  Stack,
+  Table,
+} from "../../components";
+import { Window } from "../../layouts";
+import { RadioData, RadioWires } from "./type";
+
+const formatFrequency = f => {
+  f = Math.round(f);
+  return Math.floor(f / 10) + "." + (f % 10);
+};
+
+export const Radio = (_props, context) => {
+  const { data, act } = useBackend<RadioData>(context);
+
+  const setFrequency = (value: number, finish: boolean) => {
```

Comment:
what is `finish`

Reply:
finish is set to true on the onChange handler. Handles traitor uplinks on the byond side so you cant skim through all the frequencies.